desc/magnetic_fields.py

```diff
@@ -795,3 +831,464 @@ def odefun(rpz, s):
+
+        surface current K is assumed given by
+
+        K = ∇ Φ x n
+        Φ(θ,ζ) = Φₛᵥ(θ,ζ) + Gζ/2π + Iθ/2π
+
+        where n is the winding surface unit normal
+              Phi is the current potential function,
+              which is a function of theta and zeta,
+              which is given as a secular linear term in theta/zeta
+              and a double Fourier series in theta/zeta
+        this function then uses biot-savart to find the
+        B field from this current density K on the surface
+
+    Parameters
+    ----------
+    Phi_mn : ndarray
+        Fourier coefficients of the double FourierSeries part of the current potential.
+        Should correspond to the given DoubleFourierSeries basis object passed in.
+    basis : DoubleFourierSeries
+        DoubleFourierSeries corresponding to the Phi_mn coefficients passed in.
+    I : float
+        Net current linking the plasma and the coils toroidally
+        Denoted I in the algorithm
+    G : float
+        Net current linking the plasma and the coils poloidally
+        Denoted G in the algorithm
+    surface_grid : Grid
+        grid upon which to evaluate the surface current density K
+    R_lmn, Z_lmn : array-like, shape(k,)
+        Fourier coefficients for winding surface R and Z in cylindrical coordinates
+    modes_R : array-like, shape(k,2)
+        poloidal and toroidal mode numbers [m,n] for R_lmn.
+    modes_Z : array-like, shape(k,2)
+        mode numbers associated with Z_lmn, defaults to modes_R
+    NFP : int
+        number of field periods
+    sym : bool
+        whether to enforce stellarator symmetry. Default is "auto" which enforces if
+        modes are symmetric. If True, non-symmetric modes will be truncated.
+    rho : float [0,1]
+        flux surface label for the toroidal surface
+    name : str
+        name for this surface
+    check_orientation : bool
+        ensure that this surface has a right handed orientation. Do not set to False
+        unless you are sure the parameterization you have given is right handed
+        (ie, e_theta x e_zeta points outward from the surface).
+    surface: FourierRZToroidalSurface, optional, default None
+        Existing FourierRZToroidalSurface object to create a
+        CurrentPotentialField with, if provided will use this
+        object's R_lmn, Z_lmn etc instead of any passed-in values
+
+    """
+
+    _io_attrs_ = (
+        MagneticField._io_attrs_
+        + FourierRZToroidalSurface._io_attrs_
+        + ["_surface_grid", "Phi_mn", ""]
+    )
+
+    def __init__(
+        self,
+        Phi_mn,
+        basis,
+        I,
+        G,
+        surface_grid,
+        R_lmn=None,
+        Z_lmn=None,
+        modes_R=None,
+        modes_Z=None,
+        NFP=1,
+        sym="auto",
+        rho=1,
+        name="",
+        check_orientation=True,
+        surface=None,
+    ):
+        assert isinstance(
+            basis, DoubleFourierSeries
```

Comment:
should make a public attribute for phi basis.

Reply:
Also do we want the phi basis to share M,N resolution with the underlying surface? if not we might want to make that clear and provide some way to change the resolution of the phi basis separately from the `surface.change_resolution`

Could also do what daniel did for omnigenity and have a new `M_phi` and `N_phi` to control resolution of the potential

Reply:
they are decoupled now, can see the Phi basis resolutions through `self.Phi_basis.M` (or N)